javascript/javascript3/week1/weatherApp.js

```diff
@@ -27,26 +26,23 @@ function oncheckWeatherBtnClick(e) {
   async function getWeatherData() {
     const response = await fetch(weatherURL);
     const dataWeather = await response.json();
-    document.getElementById("chosenCity").textContent += " " + dataWeather.name;
-    document.getElementById("temperature").textContent +=
-      " " + Math.round(dataWeather.main.temp) + " °C";
-    document.getElementById("windSpeed").textContent +=
-      " " + Math.round(dataWeather.wind.speed * 10) / 10 + " m/s";
+    document.getElementById("chosenCity").innerText = dataWeather.name;
+    document.getElementById("temperature").innerText =
+      Math.round(dataWeather.main.temp) + " °C";
+    document.getElementById("windSpeed").innerText =
+      Math.round(dataWeather.wind.speed * 10) / 10 + " m/s";
     document.getElementById(
       "weatherTypeIcon"
     ).src = `http://openweathermap.org/img/wn/${dataWeather.weather[0].icon}.png`;
-    document.getElementById("clouds").textContent +=
-      " " + dataWeather.clouds.all + " %";
-    document.getElementById("sunriseTime").textContent +=
-      " " + new Date(dataWeather.sys.sunrise * 1000).toLocaleTimeString();
-    document.getElementById("sunsetTime").textContent +=
-      " " + new Date(dataWeather.sys.sunset * 1000).toLocaleTimeString();
-    dataArrWeather.forEach((data) => {
-      const liItem = document.createElement("li");
-      liItem.textContent = data;
-      document.body.appendChild(liItem);
-    });
+    document.getElementById("clouds").innerText = dataWeather.clouds.all + " %";
+    document.getElementById("sunriseTime").innerText = new Date(
+      dataWeather.sys.sunrise * 1000
+    ).toLocaleTimeString();
+    document.getElementById("sunsetTime").innerText = new Date(
+      dataWeather.sys.sunset * 1000
+    ).toLocaleTimeString();
     console.log(dataWeather);
+    return response;
```

Comment:
Excellent!! Works well
You set the "onChange" event to get the city name. You can also get it in the "Click" event because the city name is only used when you press the "Search" button.
If you have another point of view, I will be happy to hear it.

Reply:
You are right. Thanks